app_dart/lib/src/service/luci.dart

```diff
@@ -170,6 +170,21 @@ class LuciService {
   ///
   /// Latest builds of each builder will be returned from new to old.
   Future<Iterable<Build>> getBuilds(String repo, bool requireTaskName, List<LuciBuilder> builders) async {
+    final List<Request> searchRequests = prepareSearchRequests(repo, requireTaskName, builders);
+    final BatchRequest batchRequest = BatchRequest(requests: searchRequests);
+    final BatchResponse batchResponse = await buildBucketClient.batch(batchRequest);
+    final Iterable<Build> builds = batchResponse.responses
+        .map<SearchBuildsResponse>((Response response) => response.searchBuilds)
+        .where((SearchBuildsResponse response) => response.builds != null)
+        .expand<Build>((SearchBuildsResponse response) => response.builds);
+    return builds;
+  }
+
+  /// Prepare search requests via tags predicate.
+  ///
+  /// For builders only enabled in prod, the scheduler_job_id follows: `flutter/<builder.name>`
+  /// For builders enabled in both prod and staging, the scheduler_job_id follows: `flutter/<realm>-<builder.name>`.
```

Comment:
Why we need new tags if we can use the bucket? ideally the predicate just adds the bucket to the query.

Reply:
That would be easy, but the Buildbucket API doesn't support bucket search: https://chromium.googlesource.com/infra/luci/luci-go/+/refs/heads/main/buildbucket/proto/builds_service.proto#381

Reply:
It supports BuilderId https://chromium.googlesource.com/infra/luci/luci-go/+/refs/heads/main/buildbucket/proto/builder.proto#15 which is a combination of project, bucket and builder name.

Reply:
I recalled why we switched from BuildId to Tags: https://github.com/flutter/cocoon/pull/1314

Dart HHH is sharing the same builder as us, and their builds miss commit sha info which is needed from our backend (https://github.com/flutter/flutter/issues/87512)